### ISSUE TITLE
Import mapillary photos

### ISSUE THREAD

**Issue by hamlet** [openstreetview/uploader#13]
Hello,
I've uploaded 440k photos to mapillary.
I don't have enough storage for that, so when they are uploaded, I got rid of them.

Could you import of my photos directly from Mapillary ? I'd be happy to share whatever credential is needed.

I don't have much bandwidth either, I don't like the idea of downloading and re-uploading half a tera-byte again.

Best regards.

**Comment by drkludge** [comment]
Unless you keep a copy of the photos, then you are out of luck.  Mapillary says that they own the photos the last time I read the FAQs.  I have saved a copy of most of my photos for myself.  I am hoping to upload them to OSV but have run into other problems.  Even then, I'll keep a copy of the photos on a USB drive.

**Comment by hamlet** [comment]
"Each Mapillary User retains all right, title and interest in and to such Mapillary User's Content, and all associated intellectual property rights, subject to these Terms" from  https://www.mapillary.com/terms
I'm not expert with this legal stuff...

**Comment by d1g** [comment]
> Unless you keep a copy of the photos, then you are out of luck. **Mapillary says that they own the photos** the last time I read the FAQs

@drkludge probably something changed since _the last time_

First, according to https://www.mapillary.com/terms  (v4, Effective Date: June 28, 2016):

> 1.4      Reservation of Rights.  
> ...
> **Each Mapillary User retains all right, title and interest in and to such Mapillary User's Content, and all associated intellectual property rights,** subject to these Terms. 
> ...

But yes, you are restricted to perform mass downloads: 
> 2.1      Specific Prohibited Uses.
> ...
> **You will not**
> (a) **use the Mapillary Solution in a manner that gives you or any other person access to mass downloads** 
> ...

Secondly, trolleyway reported that mapilarity.com interface provides an option to download source resolution, especially photo spheres [#201 Russian discussion](http://forum.openstreetmap.org/viewtopic.php?pid=607822#p607822)

So it is possible to perform 1 by 1 moves, that wouldn't be qualified as 2.1; mass downloads per user requests are not covered in their current TOS, AFAIK.

**Comment by Skippern** [comment]
I am interested in a script that can, preferably over time, download all my Mapillary sequences older than November this year as originals (not processed photos), for backup and resubmit on other projects. There are several sequences that will need heading checked, so I need a way of sorting sequences for manual handling, etc.

**Comment by seb-1204** [comment]
I would like to do similar to @Skippern to transfer 'selected' mapillary tracks to OSC using a script.

**Comment by utack** [comment]
Any updates on this?  
Asking because the upload scripts failed, and I am close to deleting all my local images

**Comment by bogdan-telenav** [comment]
HI @utack the problems is with the upload script or that you want to download images from Mapillary and after that upload on OSC?? 
Thanks 👍

**Comment by utack** [comment]
@bogdan-telenav both  
I used the GUI you have to upload now, so that is solved  
But I still have about 100.000 images on mapillary I'd eventually like to transfer some day

**Comment by adb01730** [comment]
Is there a way to upload my openstreetcam pics to mapillary?
I was using the mapillary app until if failed to detect the gps signal after a android update. I switched to OSC which I now prefer to mapillary. The OSC uploader provides % complete, time to complete, and KB/s. With Mapillary I had to monitor the upload because it would often crash after several hours and have to be restarted.

**Comment by alexandrui-telenav** [comment]
@adb01730 Good to hear your feedback! 
It is possible to copy recorded data from the OSC app before uploading, but the optimised format we're using is not compatible with Mapillary as far as we're aware. 

In order to optimise upload time, we encode the photos as frames in video files ( this reduces upload size 2-3X compared to individual JPEGs ). One could split the video files into frames and geotag them based on the metadata file before uploading to Mapillary, but it doesn't work out of the box.

**Comment by polarbearing** [comment]
@alexandrui-telenav - is that video spec identical to what the Waylens (telenav version) produces? I found metadata there as a binary, what is the structure of this binary and is it possible to convert that into a human readable form? My intention is to split the video into frames so I can review them and drop frames if necessary.

**Comment by HolgerJeromin** [comment]
> My intention is to split the video into frames so I can review them and drop frames if necessary.

Last time i checked you could change the format in the OSC app. 
I did so to be able to delete images of my hand, car dashboard, pushchair, overexposed images and other stuff.

**Comment by polarbearing** [comment]
> delete images of my hand

Yes exactly. Tried the OSC app but does not work because of Google-play dependencies (openstreetcam/openstreetview.org/issues/90, openstreetcam/android/issues/60).
Have a Waylens here however which also creates a frame-collection as a video.

**Comment by alexandrui-telenav** [comment]
@polarbearing the apps don't use the same format as Waylens, the apps only store mp4 video files + a csv metadata / track. 
I don't know how to do what you're intending to be honest, they also pack the files in a Waylens proprietary .vdb file just before upload which we decode server side. 

Question: isn't more convenient to upload everything then review & discard from the OSC website?

**Comment by signout** [comment]
It is possible for you to download the unblurred originals from mapillary using the tools from https://github.com/mapillary/mapillary_tools
You have to upload it to openstreetcam manually, but this can be done in batches if you're constrained on diskspace.

Look in the Misc section for command line options.

**Comment by signout** [comment]
To follow up on my last comment: The command in the misc section only downloads blurred downscaled images.

There is, however a tools for scraping all your images. Available at https://github.com/gitouche-sur-osm/mapillary_takeout

**Comment by hamlet** [comment]
> To finally close this there is a tool created by the community that already does this: https://osm.svimik.com/photosync/

Unfortunatly it doesn't work anymore, due to changes in the mapillary API... :-/ See https://forum.mapillary.com/t/mapillary-openstreetcam-synchronization-tool/4246/74
Too bad I missed the opportunity.

**Comment by hamlet** [comment]
> To finally close this there is a tool created by the community that already does this: https://osm.svimik.com/photosync/

Unfortunatly it doesn't work anymore, due to changes in the mapillary API... :-/ See https://forum.mapillary.com/t/mapillary-openstreetcam-synchronization-tool/4246/74
Too bad I missed the opportunity.
